File: 06-TREES/search-in-a-binary-search-tree.py
from typing import *
from TreeNode import TreeNode


# https://leetcode.com/problems/search-in-a-binary-search-tree/
class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if root is None:
            return None

        if root.val == val:
            return root

        if val < root.val:
            return self.searchBST(root.left, val)
        else:
            return self.searchBST(root.right, val)

File: 06-TREES/search-in-a-binary-search-tree.php
<?php

# https://leetcode.com/problems/search-in-a-binary-search-tree/
class Solution
{
    public function searchBTS(?TreeNode $root, int $val): ?TreeNode
    {
        if ($root === null)
            return null;

        if ($root->val == $val)
            return $root;

        if ($val < $root->val)
            return $this->searchBTS($root->left, $val);
        else
            return $this->searchBTS($root->right, $val);
    }

}
